doc/reference/ubuntu-image.rst

```diff
@@ -0,0 +1,246 @@
+
+
+Classical images
+----------------
+
+Classical images are built from a local `gadget tree`_ path.  The `gadget tree`_ is a primed `gadget snap`_ containing a ``gadget.yaml`` file in the ``meta`` directory and all the necessary bootloader gadget bits built. For instance, a `gadget tree`_ can be prepared by fetching a specially tailored `gadget snap`_ source and running ``snapcraft prime`` on it, with the resulting tree being stored in the ``prime/`` directory.
+
+
+The ``gadget.yaml`` file
+------------------------
+
+As part of the model assertion, a `gadget snap`_ is specified.  The gadget contains a ``gadget.yaml`` file, which contains the exact description of the disk-image contents in the YAML format.  The ``gadget.yaml`` file describes, among other things:
+
+* Names of all the volumes to be produced (volumes are roughly analogous to disk images).
+
+* Structures (structures define the layout of the volume, including partitions, Master Boot Records, or any other relevant content.) within the volume. :none:`wokeignore:rule=master,`
+
+* Whether the volume contains a bootloader and if so, what kind of bootloader.
+
+Note that ``ubuntu-image`` communicates with the Snap Store using the ``snap prepare-image`` sub-command.  The model-assertion file is passed to ``snap prepare-image``, which handles downloading the appropriate gadget and any extra snaps.  See the :manualpage:`snap(8) manual page <man8/snap.8.html>` for additional details.
+
+
+Basic syntax
+------------
+
+.. code-block:: yaml
+
+    ubuntu-image snap [options] model.assertion
+
+    ubuntu-image classic [options] image_definition.yaml
+
+
+General options
+~~~~~~~~~~~~~~~
+
+-h, --help
+    Show the help message and exit.
+
+--version
+    Show the program version number and exit.
+
+
+Options of the ``snap`` command
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These are the options for defining the contents of snap-based images.  They can only be used when the ``ubuntu-image snap`` command is used.
+
+``model_assertion``
+    Path to the model assertion file.  This positional argument must be given for this mode of operation.
+
+--cloud-init USER-DATA-FILE
+    ``cloud-config`` data to be copied to the image.
+
+--disable-console-conf
+    Disable ``console-conf`` on the resulting image.
+
+--factory-image
+    Hint that the image is meant to boot in a device factory.
+
+--validation=<ignore|enforce>
+    Control whether validations should be ignored or enforced.
+
+--snap SNAP
+    Install an extra snap.  This is passed through to ``snap prepare-image``. The snap argument can include additional information about the channel and/or risk with the following syntax: ``<snap>=<channel|risk>``. Note that this flag will cause an error if the model assertion has a grade higher than dangerous.
```

Comment:
This flag accepts a list of snaps.

And here also the `snap prepare-image` is now replaced.